packages/@aws-cdk/aws-eks/lib/cluster.ts

```diff
@@ -718,6 +748,123 @@ abstract class ClusterBase extends Resource implements ICluster {
+   * @param options options for adding auto scaling groups, like customizing the bootstrap script
+   */
+  public connectAutoScalingGroupCapacity(autoScalingGroup: autoscaling.AutoScalingGroup, options: AutoScalingGroupOptions) {
+    // self rules
+    autoScalingGroup.connections.allowInternally(ec2.Port.allTraffic());
+
+    // Cluster to:nodes rules
+    autoScalingGroup.connections.allowFrom(this, ec2.Port.tcp(443));
+    autoScalingGroup.connections.allowFrom(this, ec2.Port.tcpRange(1025, 65535));
+
+    // Allow HTTPS from Nodes to Cluster
+    autoScalingGroup.connections.allowTo(this, ec2.Port.tcp(443));
+
+    // Allow all node outbound traffic
+    autoScalingGroup.connections.allowToAnyIpv4(ec2.Port.allTcp());
+    autoScalingGroup.connections.allowToAnyIpv4(ec2.Port.allUdp());
+    autoScalingGroup.connections.allowToAnyIpv4(ec2.Port.allIcmp());
+
+    // allow traffic to/from managed node groups (eks attaches this security group to the managed nodes)
+    autoScalingGroup.addSecurityGroup(this.clusterSecurityGroup);
+
+    const bootstrapEnabled = options.bootstrapEnabled ?? true;
+    if (options.bootstrapOptions && !bootstrapEnabled) {
+      throw new Error('Cannot specify "bootstrapOptions" if "bootstrapEnabled" is false');
+    }
+
+    if (bootstrapEnabled) {
+      const userData = options.machineImageType === MachineImageType.BOTTLEROCKET ?
+        renderBottlerocketUserData(this) :
+        renderAmazonLinuxUserData(this, autoScalingGroup, options.bootstrapOptions);
+      autoScalingGroup.addUserData(...userData);
+    }
+
+    autoScalingGroup.role.addManagedPolicy(iam.ManagedPolicy.fromAwsManagedPolicyName('AmazonEKSWorkerNodePolicy'));
+    autoScalingGroup.role.addManagedPolicy(iam.ManagedPolicy.fromAwsManagedPolicyName('AmazonEKS_CNI_Policy'));
+    autoScalingGroup.role.addManagedPolicy(iam.ManagedPolicy.fromAwsManagedPolicyName('AmazonEC2ContainerRegistryReadOnly'));
+
+    // EKS Required Tags
+    // https://docs.aws.amazon.com/eks/latest/userguide/worker.html
+    Tags.of(autoScalingGroup).add(`kubernetes.io/cluster/${this.clusterName}`, 'owned', {
+      applyToLaunchedInstances: true,
+      // exclude security groups to avoid multiple "owned" security groups.
+      // (the cluster security group already has this tag)
+      excludeResourceTypes: ['AWS::EC2::SecurityGroup'],
+    });
+
+    // do not attempt to map the role if `kubectl` is not enabled for this
+    // cluster or if `mapRole` is set to false. By default this should happen.
+    let mapRole = options.mapRole ?? true;
+    if (mapRole && !this._awsAuth) {
+      Annotations.of(autoScalingGroup).addWarning('Auto-mapping aws-auth role for imported cluster is not supported, please map role manually');
+      mapRole = false;
+    }
+    if (mapRole) {
+      // see https://docs.aws.amazon.com/en_us/eks/latest/userguide/add-user-role.html
+      this._awsAuth?.addRoleMapping(autoScalingGroup.role, {
```

Comment:
The only reason `this._awsAuth` would be initialized in the case of a managed cluster is because we happen to call `this.awsAuth` in the cluster initialization:

https://github.com/aws/aws-cdk/blob/8ed088a4cc5388b85776c3486cb657462836d923/packages/%40aws-cdk/aws-eks/lib/cluster.ts#L1109

https://github.com/aws/aws-cdk/blob/8ed088a4cc5388b85776c3486cb657462836d923/packages/%40aws-cdk/aws-eks/lib/cluster.ts#L1319-L1325

This feels fragile. Instead i'd rather we introduce:

`public abstract readonly awsAuth` 

in `ClusterBase`, and have `ImportedCluster` throw an exception when it's called.